system/modules/core/library/Contao/Encryption.php

```diff
@@ -185,61 +185,25 @@ protected static function initialize()
 	 */
 	public static function hash($strPassword)
 	{
-		$intCost = \Config::get('bcryptCost') ?: 10;
+		@trigger_error('Using Encryption::hash() has been deprecated and will no longer work in Contao 5.0. Use password_hash() instead.', E_USER_DEPRECATED);
 
-		if ($intCost < 4 || $intCost > 31)
-		{
-			throw new \Exception("The bcrypt cost has to be between 4 and 31, $intCost given");
-		}
-
-		if (function_exists('password_hash'))
-		{
-			return password_hash($strPassword, PASSWORD_BCRYPT, array('cost'=>$intCost));
-		}
-		elseif (CRYPT_BLOWFISH == 1)
-		{
-			return crypt($strPassword, '$2y$' . sprintf('%02d', $intCost) . '$' . md5(uniqid(mt_rand(), true)) . '$');
-		}
-		elseif (CRYPT_SHA512 == 1)
-		{
-			return crypt($strPassword, '$6$' . md5(uniqid(mt_rand(), true)) . '$');
-		}
-		elseif (CRYPT_SHA256 == 1)
-		{
-			return crypt($strPassword, '$5$' . md5(uniqid(mt_rand(), true)) . '$');
-		}
-
-		throw new \Exception('None of the required crypt() algorithms is available');
+		return password_hash($strPassword, PASSWORD_DEFAULT);
 	}
 
 
 	/**
-	 * Test whether a password hash has been generated with crypt()
+	 * Test whether a password hash has been generated with a password API compatible
+	 * algorithm.
 	 *
 	 * @param string $strHash The password hash
 	 *
-	 * @return boolean True if the password hash has been generated with crypt()
+	 * @return boolean True if compatible
 	 */
 	public static function test($strHash)
 	{
-		if (strncmp($strHash, '$2y$', 4) === 0)
-		{
-			return true;
-		}
-		elseif (strncmp($strHash, '$2a$', 4) === 0)
-		{
-			return true;
-		}
-		elseif (strncmp($strHash, '$6$', 3) === 0)
-		{
-			return true;
-		}
-		elseif (strncmp($strHash, '$5$', 3) === 0)
-		{
-			return true;
-		}
+		$info = password_get_info($strHash);
 
-		return false;
+		return 0 !== $info['algo'];
```

Comment:
There is no constant for unknown password algorithms :(

Reply:
Why not `$info['algo'] > 0`?

Reply:
It's the same? I don't really care :D